### ISSUE TITLE
Pyright thinks that star imports overwrite the module docstring

### ISSUE THREAD

**Issue by tmke8** [microsoft/pyright#5737]
Given this file layout:

```
mod
├── __init__.py
└── submod
    ├── __init__.py
    └── a.py
```

with these contents:

`mod/__init__.py`:
```python
"""Docstring."""
from .submod import *
```

`mod/submod/__init__.py`:
```python
from .a import *
```

`mod/submod/a.py`:
```python
__all__ = ["f"]

def f(x: int) -> None:
    print(x)
```

I get this error:
```
$ pyright mod
/home/tmke8/dev/python/playground/mod/__init__.py
  /home/tmke8/dev/python/playground/mod/__init__.py:2:21 - error: Import symbol "__doc__" has type "str | None", which cannot be assigned to declared type "str"
    Type "str | None" cannot be assigned to type "str"
      Type "None" cannot be assigned to type "str" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations
```

It seems that pyright sees that `mod/submod/__init__.py` doesn't have a `__doc__` so it's not compatible with the `__doc__` from `mod/__init__.py`.

This appeared with pyright 1.1.323 but I didn't see anything about this in the release notes.

**Comment by erictraut** [comment]
Version 1.1.323 includes a new feature: `Enhancement: Added type compatibility enforcement for symbols imported via a wildcard import.`. 

The issue you're reporting here is a bug that predated the new feature, but it was not previously surfaced in a way that anyone noticed. The issue stems from an oversight of a subtle exemption in the [Python documentation](https://docs.python.org/3/reference/simple_stmts.html#the-import-statement). It says that a wildcard import targeting a module with no `__all__` should include all names found in the module's namespace _which do not begin with an underscore character_. I mistakenly interpreted that to mean that symbols like `_foo` and `_T` should be exempted but that dundered names should still be included. It turns out that this exemption also applies to dundered symbols (like `__doc__`).

This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.324, which I just published. It will also be included in a future release of pylance.
